### PR TITLE
CTECH-691: First draft

### PR DESCRIPTION
**Pull request by lorenzdec**

### CODE REVIEW

File: examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb [hunk 1]
```diff
@@ -0,0 +1,747 @@
+{
```
Comment (cscialino) [inline examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb:1]: <p>Wondering if this should be a more general feature on storing any external document in the SRS. The use-case in the example would then be the portfolio-level data.</p><p>Perhaps it should then also reside in&nbsp;<code>sample-notebooks/examples/features/core-lusid/</code>, rather then the valuation folder.</p><p>Also could you add the usual docstring as the first cell as stated in the notebook style-guide?</p><p><a href="https://github.com/finbourne/sample-notebooks/blob/07981d1ce6deb4a1fd752d70c78b08c223a11dd0/docs/FINBOURNE%20notebook%20style%20guide.ipynb" target="_blank">sample-notebooks/FINBOURNE notebook style guide.ipynb at 07981d1ce6deb4a1fd752d70c78b08c223a11dd0 · finbourne/sample-notebooks (github.com)</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (cscialino) [inline examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb:1]: <p>Could we also add some description to what the data ID is, some remarks on the versioning (I believe the documents are immutable, meaning updates require a bump of the version, etc.).</p><p>Might be good to add a quick link back to the API docs while describing it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (cscialino) [inline examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #16.        pf_df.to_csv(s)   
</pre><p>Could we add some code comments in this cell overall, just to add a little description on what's going on.</p><p>Also, do we need StringIO, or could the df not simply be stored as a csv directly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (cscialino) [inline examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb:1]: <p>Looks good, I'm curious if this data is somehow accessible via something like&nbsp;<em>GetValuation</em>.</p><p>It seems like in this use-case we simply store a table within the SRS, and then play it back.</p><p>That might beg the question why we are creating the portfolios in the setup piece?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/412/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (lorenzdec) [reply examples/use-cases/valuation/Externally Calculated Portfolio Level Data.ipynb:1]: <p>the portfolios are created to give context I suppose, as in theory we are simply uploading and returning a table from the SRS. But in isolation that might seem a bit weird to a client imo, so creating the portfolios which are then mentioned later in the data table gives it a sort of thread throughout the notebook</p><p> </p>

**Comment by cscialino** [comment]
Looks like the build is failing due to the notebook, it looks like you may have a missing data file. Perhaps you still need to move it over to the core-lusid directory, looks like it may be this one:

[examples/use-cases/valuation/data/srs_portfolio_level_example_portfolios.csv](https://github.com/finbourne/sample-notebooks/pull/412/files#diff-c2b87878ac89390af5e4fd837699639f284875b024545838aeb6ef9b1835fe76)

**Pull request merged**